stack/stack.toml

```diff
@@ -0,0 +1,63 @@
+id = "io.buildpacks.stacks.jammy"
+homepage = "https://github.com/paketo-buildpacks/jammy-base-stack"
+maintainer = "Paketo Buildpacks"
+
+platforms = ["linux/amd64"]
+
+[build]
+  description = "ubuntu:jammy with compilers and shell utilities"
+  dockerfile = "./build.Dockerfile"
+  gid = 1000
+  shell = "/bin/bash"
+  uid = 1001
+
+  [build.args]
+    sources = """
+    deb http://archive.ubuntu.com/ubuntu jammy main
+    deb http://archive.ubuntu.com/ubuntu jammy-updates main
+    deb http://archive.ubuntu.com/ubuntu jammy-security main
+    """
+
+    packages = """\
+    build-essential \
+    ca-certificates \
+    curl \
+    git \
+    jq \
+    libexpat1 \
+    libgmp-dev \
+    libssl3 \
+    libyaml-0-2 \
+    netbase \
+    openssl \
+    tzdata \
+    xz-utils \
+    zlib1g-dev \
+    """
+
+[run]
+  description = "ubuntu:jammy with some common dependencies like tzdata and openssl"
+  dockerfile = "./run.Dockerfile"
+  gid = 1000
+  shell = "/bin/bash"
+  uid = 1002
+
+  [run.args]
+    sources = """
+    deb http://archive.ubuntu.com/ubuntu jammy main
+    deb http://archive.ubuntu.com/ubuntu jammy-updates main
+    deb http://archive.ubuntu.com/ubuntu jammy-security main
+    """
+
+    packages = """\
+    base-files \
+    ca-certificates \
+    libc6 \
```

Comment:
I think this is a addition to this package list that is different than what we have in bionic. Can you explain why we need to include this package here?

Reply:
Additionally, we prefer the packages in the run image be a subset of the packages for in the build image, so I'd also like to understand if there's rationale for having it in the run image only.

Reply:
We took that from https://github.com/paketo-buildpacks/jammy-tiny-stack/blob/4d348c6c13b1b6c815c637a5f824f05ec367c06a/stack/stack.toml#L48

Our reasoning was that everything that is part of the `tiny` stack should also be part of the `base` stack.

Reply:
`libc6` is already present on ubuntu. We explicitly add it to `tiny`'s `run` image because that is distroless.

I think it's clearer to remove it from the list.

Verifying that it is already present on `ubuntu:jammy`:
```
 ❯ docker run -it ubuntu:jammy /bin/bash
root@6dea59df0ae8:/# echo "deb http://archive.ubuntu.com/ubuntu jammy main" > /etc/apt/sources.list
root@6dea59df0ae8:/# echo "deb http://archive.ubuntu.com/ubuntu jammy-updates main" >> /etc/apt/sources.list
root@6dea59df0ae8:/# echo "deb http://archive.ubuntu.com/ubuntu jammy-security main" >> /etc/apt/sources.list
root@6dea59df0ae8:/# apt update && apt install libc6
Get:1 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]
Get:2 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [109 kB]
Get:3 http://archive.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
Get:4 http://archive.ubuntu.com/ubuntu jammy/main amd64 Packages [1792 kB]
Get:5 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [406 kB]
Get:6 http://archive.ubuntu.com/ubuntu jammy-security/main amd64 Packages [238 kB]
Fetched 2926 kB in 2s (1842 kB/s)
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
8 packages can be upgraded. Run 'apt list --upgradable' to see them.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
libc6 is already the newest version (2.35-0ubuntu3).
0 upgraded, 0 newly installed, 0 to remove and 8 not upgraded.
```

Reply:
Looking at the [Bionic Base stack](https://github.com/paketo-buildpacks/base-stack-release/blob/main/build-receipt), it looks like `libc6` is on the build and run images, even though it's not explicitly required in the Bionic Base stack package list. 

@phil9909 I think your rationale to keep `tiny` as a subset of `base` makes sense and is a guarantee we should try to uphold. I would like to understand if we should include this package in the Base build image as well for consistency's sake. thoughts @robdimsdale @ryanmoran

Reply:
I think we should evaluate the "package list subset claim" on the resultant images, not the input in `stack.toml`.

I say this because the OS brings in a bunch of packages that we don't explicitly list (and probably shouldn't).

That being said, for everything except `tiny`/`run`, we use the same base image, which therefore will have the same set of packages, so if we treat `tiny`/`run` as a special case, then we can consider a subset of the package list in `stack.toml` as equivalent to a subset of the resultant package list, as long as we are consistent in not re-including packages that ship with the OS, such as `libc6`.

Reply:
Here is a list of all the packages that ship by default with ubuntu jammy:

```
 ❯ docker run -it ubuntu:jammy /bin/bash
root@48163cbb2bb8:/#  dpkg -l | cut -d' ' -f3
adduser
apt
base-files
base-passwd
bash
bsdutils
coreutils
dash
debconf
debianutils
diffutils
dpkg
e2fsprogs
findutils
gcc-12-base:amd64
gpgv
grep
gzip
hostname
init-system-helpers
libacl1:amd64
libapt-pkg6.0:amd64
libattr1:amd64
libaudit-common
libaudit1:amd64
libblkid1:amd64
libbz2-1.0:amd64
libc-bin
libc6:amd64
libcap-ng0:amd64
libcap2:amd64
libcom-err2:amd64
libcrypt1:amd64
libdb5.3:amd64
libdebconfclient0:amd64
libext2fs2:amd64
libffi8:amd64
libgcc-s1:amd64
libgcrypt20:amd64
libgmp10:amd64
libgnutls30:amd64
libgpg-error0:amd64
libgssapi-krb5-2:amd64
libhogweed6:amd64
libidn2-0:amd64
libk5crypto3:amd64
libkeyutils1:amd64
libkrb5-3:amd64
libkrb5support0:amd64
liblz4-1:amd64
liblzma5:amd64
libmount1:amd64
libncurses6:amd64
libncursesw6:amd64
libnettle8:amd64
libnsl2:amd64
libp11-kit0:amd64
libpam-modules:amd64
libpam-modules-bin
libpam-runtime
libpam0g:amd64
libpcre2-8-0:amd64
libpcre3:amd64
libprocps8:amd64
libseccomp2:amd64
libselinux1:amd64
libsemanage-common
libsemanage2:amd64
libsepol2:amd64
libsmartcols1:amd64
libss2:amd64
libssl3:amd64
libstdc++6:amd64
libsystemd0:amd64
libtasn1-6:amd64
libtinfo6:amd64
libtirpc-common
libtirpc3:amd64
libudev1:amd64
libunistring2:amd64
libuuid1:amd64
libxxhash0:amd64
libzstd1:amd64
login
logsave
lsb-base
mawk
mount
ncurses-base
ncurses-bin
passwd
perl-base
procps
sed
sensible-utils
sysvinit-utils
tar
ubuntu-keyring
usrmerge
util-linux
zlib1g:amd64
```

Reply:
@robdimsdale I hadn't seen your comment when I posted mine-- what you wrote makes sense.

> I think we should evaluate the "package list subset claim" on the resultant images, not the input in stack.toml.

^ +1 on this. 

So the outcome here for @phil9909 is that we should remove `libc6` from the `stack.toml` here, since it will be implicitly included with ubuntu. It's only added to Tiny run image explicitly because the tiny run image is based on the `scratch` image, not `ubuntu:jammy`